Fixes the FontSizePicker Custom option

JeanDavidDaviet opened the pull request:
Fixes #18716 
This commit prevent the "Custom" option still showing up in the font sizes drop down when  a WordPress theme uses the `add_theme_support( 'disable-custom-font-sizes' )` feature.

## Description
There was no check on the `disableCustomFontSizes`variable. The custom object `{ slug: 'custom', name: __( 'Custom' ) }` was then always added even when not wanted.

## How has this been tested?
Tested with Storybook and npm test

## Types of changes
Bug Fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.

JeanDavidDaviet committed:
Fixes the FontSizePicker Custom option

This commit prevent the "Custom" option still showing up in the font sizes drop down when  a WordPress theme uses the `add_theme_support( 'disable-custom-font-sizes' )` feature.

ZebulanStanphill commented on packages/components/src/font-size-picker/index.js:
```diff
@@ -21,14 +21,19 @@ function getSelectValueFromFontSize( fontSizes, value ) {
+			{ slug: 'custom', name: __( 'Custom' ) },
+		];
+	}
 	return [
```
The return statement is over-complicated since you can just return `optionsArray.map(...)` directly. There's no need to spread the array into another array.

ZebulanStanphill commented on packages/components/src/font-size-picker/index.js:
```diff
@@ -21,14 +21,19 @@ function getSelectValueFromFontSize( fontSizes, value ) {
-function getSelectOptions( optionsArray ) {
+function getSelectOptions( optionsArray, disableCustomFontSizes ) {
+	if ( ! disableCustomFontSizes ) {
+		optionsArray = [
```
I think this is a bit confusing to read, since at first glance it looks like you are mutating the original array, when in fact you are only changing the function-scoped variable.

Also, I haven't tested the code yet, but wouldn't adding the custom value here cause problems with the `map` callback, which tries to get the `size` property of every `option`? In any case, it seems a bit confusing.

JeanDavidDaviet replied:
I don't get what you mean about the mutating part ?
The map function goes over the custom option too, but as the custom size needs to be undefined in `getSelectValueFromFontSize` above, it works. I guess it's not optimal, I would like some suggestion on how to improve this piece of code but trying also to be DRY.

ZebulanStanphill replied:
Well, you're not suppose to mutate props/state directly in React components, and setting `optionsArray` kind of looks like a mutation of the original value passed to `getSelectOptions`. It's **not** actually a mutation, since you're only replacing the local variable's reference value with a reference to an entirely new array; if you had used something like `optionsArray.push()` that _would_ have mutated the array. I just thought it looked a little confusing on first glance.

Good point about how `getSelectValueFromFontSize` works. On second examination, I guess rewriting `getSelectOptions` without introducing unnecessary repetition would be harder than I thought. Nevermind. :slightly_smiling_face:

JeanDavidDaviet committed:
Return directly the optionArray for the FontSizePicker instead of spread

JeanDavidDaviet committed:
Update the fontSizePicker when font size slider is used

The slider didn't update the select options. In the new `onSliderChangeValue` function, we set the new fontSize state with the props function `onChange` and we also set the selection option given the new font size.

JeanDavidDaviet committed:
Revert "Update the fontSizePicker when font size slider is used"

This reverts commit 73f952f33e22723718b73f5cb2ce4cf90085cae4.
I don't know why Travis doesn't accept this commit.

JeanDavidDaviet reviewed (commented):
I don't understand why this change makes Travis build fail. @ZebulanStanphill ?
They are necessary to update the custom select when the fontsize slider is used.

ZebulanStanphill commented:
@JeanDavidDaviet The automated tests haven't been working quite right lately, so the failure may be unrelated. Try recommitting the change.

JeanDavidDaviet committed:
Update the fontSizePicker when font size slider is used

This reverts commit aa626435c519b76688b33e2556b01036f823082b.

JeanDavidDaviet commented:
@ZebulanStanphill Thanks, that was it !

epiqueras commented:
It would be good to add tests for this.

JeanDavidDaviet commented:
> It would be good to add tests for this.

@epiqueras I won't be home for december, but I sure would like to see the test suite for this feature implemented (or modified) as I'm not a good test writer.
